### PR TITLE
Implement `--dry-run` for rich console

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
See https://github.com/gradle/gradle/issues/2060

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Branches_Checkpoints&branch_Gradle_Branches_Checkpoints=dl%2Frich-dry-run-better)
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/execution/DryRunBuildExecutionActionTest.java [hunk 1]
```diff
@@ -1,84 +0,0 @@
-/*
```
Comment (eriwen) [inline subprojects/core/src/test/groovy/org/gradle/execution/DryRunBuildExecutionActionTest.java]: 👏  Bye bye JMock!

**Comment by alkemist** [comment]
Should test for buildSrc behaviour too.

**Comment by lacasseio** [comment]
@eriwen Don't worry about it I spent lots of hours doing a mess when the actual solution from @adammurdoch (which should get all the credit for) was really simple.

**Comment by lacasseio** [comment]
This PR was merged but doesn't show as merge. Closing it now.

**Pull request closed without merging**